<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
        <title>ATAC-Seq Analysis Pipeline</title>
        <meta name="viewport" content="width=device-width">

        <!-- syntax highlighting CSS -->
        <link rel="stylesheet" href="/css/syntax.css">

        <!-- Custom CSS -->
        <link rel="stylesheet" href="/css/main.css">

    </head>
    <body>

        <div class="site">
          <div class="header">
            <h1 class="title"><a href="/">Orolab</a></h1>
            <a class="extra" href="/">home</a>
          </div>

          <h2>ATAC-Seq Analysis Pipeline</h2>
<p class="meta">04 Aug 2014</p>

<div class="post">
<p>Generally, the pipeline includes the following steps:</p>

<ul>
<li>Align the reads to reference genome by <a href="http://bowtie-bio.sourceforge.net/index.shtml">Bowtie</a></li>
<li>Make UCSC track</li>
<li>Call peaks by <a href="http://liulab.dfci.harvard.edu/MACS/">MACS</a> (use macs2 instead of macs14)</li>
<li>Annotating peaks by <a href="http://biowhat.ucsd.edu/homer/ngs/">HOMER</a></li>
</ul>

<p><strong>Note</strong>, You will need two additional scripts. One is called <a href="https://github.com/riverlee/ATAC/blob/master/code/getFragmentBed.pl">getFragmentBed.pl</a> which will convert the mapping coordinates of each mate to the biological fragment level; 
the other is called <a href="https://github.com/riverlee/ATAC/blob/master/code/preShift.pl">preShift.pl</a> which shifts the reads mapping to plus strand upstream <strong><em>n bp</em></strong> and reads mapping to the minus strand downstream <strong><em>n bp</em></strong>.</p>

<div class="highlight"><pre><code class="bash"><span class="c">#!/bin/bash</span>
<span class="c">###################################</span>
<span class="c"># Author: Jiang Li</span>
<span class="c"># Email:  riverlee2008@gmail.com</span>
<span class="c">###################################</span>
<span class="c"># ATAC-Seq analysis pipeline</span>
<span class="c">###################################</span>


<span class="c">###################################</span>
<span class="c"># An example of ATAC-Seq analysis</span>
<span class="c"># Suppose your input file is R1.fq and R2.fq</span>
<span class="c">###################################</span>
<span class="c">###################################</span>
<span class="c"># Alignment</span>
<span class="c">###################################</span>
<span class="c"># 1.1 Alignment </span>
<span class="nb">echo</span> <span class="s1">&#39;bowtie alignment&#39;</span> <span class="sb">`</span>date<span class="sb">`</span>
bowtie --chunkmbs 256 -p 24 -S  -m 1 -X 2000  -t hg19  R1.fq R2.fq my.sam
samtools view -bS  my.sam |samtools sort - my

<span class="c"># 1.2 Sam to bam and only output those mapped</span>
samtools view -b -F my.bam &gt;my_norm_sorted.bam

<span class="c"># 1.3 Delete not sorted bam</span>
<span class="nb">echo</span> <span class="s2">&quot;delete bam&quot;</span> <span class="sb">`</span>date<span class="sb">`</span>
rm -rf my.sam

<span class="c"># 1.4 remove duplicates and them remove chrM reads</span>
<span class="nb">echo</span> <span class="s2">&quot;remove duplicates&quot;</span> <span class="sb">`</span>date<span class="sb">`</span>
samtools rmdup  my_norm_sorted.bam my_norm_sorted_rmdup.bam
<span class="nb">echo</span> <span class="s2">&quot;Remove chrM reads&quot;</span> <span class="sb">`</span>date<span class="sb">`</span>
samtools view -h my_norm_sorted_rmdup.bam |perl -lane <span class="s1">&#39;print $_ if $F[2] ne &quot;chrM&quot;&#39;</span> |samtools view -bS - &gt;my_norm_sorted_rmdup_nochrM.bam
samtools index my_norm_sorted_rmdup_nochrM.bam

<span class="c"># 1.5 Make Fragment Bed </span>
getFragmentBed.pl my_norm_sorted_rmdup_nochrM.bam my_fragment.bed

<span class="c">###################################</span>
<span class="c"># Make UCSC Tracks</span>
<span class="c">###################################</span>
<span class="c"># 2.1 make bedGraph</span>
<span class="nb">echo</span> <span class="s2">&quot;make bedgraph&quot;</span> <span class="sb">`</span>date<span class="sb">`</span>
sort -k1,1 -k2,2n my_fragment.bed |bedItemOverlapCount hg19 -chromSize<span class="o">=</span>hg19.chrom.sizes stdin |sort -k1,1 -k2,2n &gt;my.bedGraph

<span class="c"># 2.2 make bigwig</span>
bedGraphToBigWig my.bedGraph hg19.chrom.sizes my.bw


<span class="c">###################################</span>
<span class="c"># Call peaks</span>
<span class="c"># Before appling macs2, pre-shift the reads</span>
<span class="c">###################################</span>
preShiftBed.pl my_fragment.bed 75 my_fragment_preShift75.bed

<span class="nb">test</span> ! -r callpeaks <span class="o">&amp;&amp;</span> mkdir callpeaks
<span class="nb">cd </span>callpeaks
macs2 callpeak  -t ../my_fragment_preShift75.bed -f BED  -g hs -n my --nomodel --shiftsize 75

<span class="c">###################################</span>
<span class="c"># Peak annotation by Homme</span>
<span class="c">###################################</span>
annotatePeaks.pl my_peaks.narrowPeak hg19 &gt; my_peaks_anno.txt
</code></pre>
</div>

</div>


          <div class="footer">
            <div class="contact">
              <p>
                Jiang Li<br />
                Bioinformatician at Stanford<br />
              </p>
            </div>
            <div class="contact">
              <p>
                <a href="https://github.com/riverlee">github.com/riverlee</a><br />
                <a href="mailto:jiangli@stanford.edu">jiangli@stanford.edu</a>
              </p>
            </div>
          </div>
        </div>
    <a href="http://github.com/riverlee/orolab"><img style="position: absolute; top: 0; right: 0; border: 0;" src="http://s3.amazonaws.com/github/ribbons/forkme_right_red_aa0000.png" alt="Fork me on GitHub" /></a>

    </body>
</html>
